frontend/src/app/auth/auth.service.ts

```diff
@@ -4,21 +4,28 @@ import 'rxjs/add/operator/do';
 import 'rxjs/add/operator/delay';
 import {HttpClient, HttpHeaders} from '@angular/common/http';
 import {Oauth} from './oauth';
+import {ConfigService} from "../config/config.service";
+import {Config} from "../config/config";
 
 
 
 @Injectable()
 export class AuthService {
 
-    constructor(private http: HttpClient) { }
+    config: Config;
+    tokenURL: string;
+
+    constructor(private http: HttpClient, private configService: ConfigService) {
+        this.config = configService.getConfig();
+        this.tokenURL = 'http://localhost:' + this.config.backendPort + '/oauth/token';
```

Comment:
Why localhost and not a config item too? This won’t work in production, the browser needs to hit test.h2ms.org.